http_2.rflx

```diff
@@ -0,0 +1,230 @@
+							Frame_Type = PING or
+							Frame_Type = CONTINUATION or
+							Frame_Type = GOAWAY or
+							Frame_Type = RST_STREAM or
+							Frame_Type = WINDOW_UPDATE or
+							Frame_Type = PRIORITY
+						) and Flag_Padded = False));
+			Bit_5_Unused : Zero_Bit;
+			Flag_Priority : Boolean
+				then Bit_7_Unused
+					if (Frame_Type = HEADERS or
+						((	Frame_Type = DATA or
+							Frame_Type = SETTINGS or
+							Frame_Type = PING or
+							Frame_Type = GOAWAY or
+							Frame_Type = RST_STREAM or
+							Frame_Type = WINDOW_UPDATE or
+							Frame_Type = PRIORITY or
+							Frame_Type = PUSH_PROMISE or
+							Frame_Type = CONTINUATION
+						) and Flag_Priority = False));
+			Bit_7_Unused : Zero_Bit;
+			Bit_8_Unused : Zero_Bit;
+			Reserved_Bit_1 : Zero_Bit;
+			Stream_Identifier : Stream_ID 	-- Last Header Field
+				then Pad_Length
+					if (Frame_Type = DATA or Frame_Type = HEADERS or Frame_Type = PUSH_PROMISE) and Flag_Padded = True
+				then Data
+					with Size => Payload_Length * 8
+					if Frame_Type = DATA and Flag_Padded = False
+				then Exclusive_Flag
+					if Frame_Type = HEADERS and Flag_Priority = True and Flag_Padded = False
+				then Header_Block_Fragment
+					with Size => Payload_Length * 8
+					if (Frame_Type = HEADERS and Flag_Priority = False and Flag_Padded = False) or Frame_Type = CONTINUATION
+				then Exclusive_Flag
+					if Frame_Type = PRIORITY
+				then Error_Code
+					if Frame_Type = RST_STREAM
+				then Settings_Identifier
+					if Frame_Type = SETTINGS
+				then Data
+					with Size => 64
+					if Frame_Type = PING and Stream_Identifier = 0 -- RFC: If a PING frame is received with a stream identifier field value other than 0x0, the recipient MUST respond with a connection error (Section 5.4.1) of type PROTOCOL_ERROR.
+				then Reserved_Bit_2
+					if (Frame_Type = GOAWAY and Stream_Identifier = 0) or
+						Frame_Type = WINDOW_UPDATE or
+						(Frame_Type = PUSH_PROMISE and Flag_Padded = False);
+			Settings_Identifier : Settings_Parameter
+				then Settings_Value_Enable_Push
+					if Settings_Identifier = SETTINGS_ENABLE_PUSH
+				then Settings_Value_Initial_Window_Size
+					if Settings_Identifier = SETTINGS_INITIAL_WINDOW_SIZE
+				then Settings_Value_Max_Frame_Size
+					if Settings_Identifier = SETTINGS_MAX_FRAME_SIZE
+				then Settings_Value_Generic
+					if (Settings_Identifier = SETTINGS_HEADER_TABLE_SIZE or
+						Settings_Identifier = SETTINGS_MAX_CONCURRENT_STREAMS or
+						Settings_Identifier = SETTINGS_MAX_HEADER_LIST_SIZE);
+			Settings_Value_Enable_Push : Settings_Value_Enable_Push
+				then null;
+			Settings_Value_Initial_Window_Size : Settings_Value_Initial_Window_Size
+				then null;
+			Settings_Value_Max_Frame_Size : Settings_Value_Max_Frame_Size
+				then null;
+			Settings_Value_Generic : Settings_Value_Generic
+				then null;
+			Pad_Length : Pad_Length
+				then Data
+					with Size => (Payload_Length - (Pad_Length + 1)) * 8 	-- 1 -> Pad_Length'Length
```

Comment:
Instead of the comment `Pad_Length'Size` should be used in the expression.

Reply:
Changed in d40f173